miden-crypto/src/merkle/forest/mod.rs

```diff
@@ -0,0 +1,130 @@
+use alloc::{collections::BTreeSet, collections::BTreeMap, vec::Vec};
+use core::borrow::Borrow;
+
+use super::{
+    MerkleStore, SmtLeaf, SmtProof,
+    EmptySubtreeRoots, InnerNodeInfo, MerkleError, MerklePath, MerkleProof, MerkleTree, NodeIndex,
+    PartialMerkleTree, RootPath, Rpo256, SimpleSmt, Smt, Word, mmr::Mmr,
+};
+use crate::{
+    merkle::empty_roots, utils::{ByteReader, ByteWriter, Deserializable, DeserializationError, Serializable}, Map
+};
+
+#[cfg(test)]
+mod tests;
+
+// SPARSE MERKLE TREE FOREST
+// ================================================================================================
+
+
+/// An in-memory data store for TODO
+///
+/// TODO description.
+///
+/// Example usage:
+///
+/// ```rust
+///   TODO
+/// ```
+#[derive(Debug, Clone, Eq, PartialEq)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
+pub struct SmtForest {
+    /// Roots of all SMTs in this forest. Any time an SMT in this forest is updated, we add a new
+    /// root to this set.
+    roots: BTreeSet<Word>,
+
+    /// Stores Merkle paths for all SMTs in this forest.
+    store: MerkleStore,
+
+    /// Leaves of all SMTs stored in this forest
+    leaves: BTreeMap<Word, SmtLeaf>,
+}
+
+impl Default for SmtForest {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+const DEPTH: u8 = 64;
+
+impl SmtForest {
+    // CONSTRUCTORS
+    // --------------------------------------------------------------------------------------------
+
+    /// Creates an empty `SmtForest` instance.
+    pub fn new() -> SmtForest {
+        let store = MerkleStore::new();
+        let mut roots = BTreeSet::new();
+        // TODO not sure about 0 - it should be an empty tree root
+        // TODO have a test for new() that checks this
+        let empty_tree_root = EmptySubtreeRoots::empty_hashes(DEPTH)[0];
```

Comment:
You could also use `EmptySubtreeRoots::entry(tree_depth, root_depth = 0)` to avoid dealing with slices.